R/comp_report.R

```diff
@@ -57,33 +57,31 @@ comp_report = function(this_sample_id,
     colnames(maf)[end_col_maf] = "End_Position"
   }
 
-  if(!missing(seq_data)){
-    seq = seq_data
-    seq = as.data.frame(seq)
-    colnames(seq)[chrom_col_seq] = "chrom"
-    colnames(seq)[start_col_seq] = "start"
-    colnames(seq)[end_col_seq] = "end"
-    colnames(seq)[cn_col_seq] = "CN"
+  if(!missing(seg_data)){
+    seg = seg_data
+    seg = as.data.frame(seg)
+    colnames(seg)[chrom_col_seg] = "chrom"
+    colnames(seg)[start_col_seg] = "start"
+    colnames(seg)[end_col_seg] = "end"
+    colnames(seg)[cn_col_seg] = "CN"
 
-  }else if (!is.null(seq_path)){
-    seq = read.table(seq_path, sep = "\t", header = TRUE, row.names = FALSE, col.names = FALSE, quote = FALSE)
-    seq = as.data.frame(seq)
-    colnames(seq)[chrom_col_seq] = "chrom"
-    colnames(seq)[start_col_seq] = "start"
-    colnames(seq)[end_col_seq] = "end"
-    colnames(seq)[cn_col_seq] = "CN"
+  }else if (!is.null(seg_path)){
+    seg = read.table(seg_path, sep = "\t", header = TRUE, row.names = FALSE, col.names = FALSE, quote = FALSE)
+    seg = as.data.frame(seg)
+    colnames(seg)[chrom_col_seg] = "chrom"
+    colnames(seg)[start_col_seg] = "start"
+    colnames(seg)[end_col_seg] = "end"
+    colnames(seg)[cn_col_seg] = "CN"
   }
 
-  #read maf and seq data into r (avoid calling assign_cn_to_ssm and get_cn_segments for every plotting function)
+  #read maf and seg data into r (avoid calling assign_cn_to_ssm and get_cn_segments for every plotting function)
   if(missing(maf_data) && is.null(maf_path)){
-    maf = assign_cn_to_ssm(
-      this_sample_id = this_sample_id,
-      coding_only = FALSE,
-      this_seq_type = this_seq_type)$maf
+    maf = get_ssm_by_sample(these_sample_ids = this_sample_id, 
+                            projection = projection)
```

Comment:
The object `projection` is missing. Is this a parameter of `comp_report`?

Reply:
Good catch, I will fix this! Thanks